src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -240,23 +298,182 @@ inline DocumentStatisticsAnnotationData::DocumentStatisticsAnnotationData(Freque
 
 // ----------------------------------------------------------------------
 // |
-// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |  DocumentStatisticsEstimator
 // |
 // ----------------------------------------------------------------------
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(void) :
-    _totalNumDocuments(0) {
+//Would it be better to place default values during pipeline chaining(countvectorizer and tfidfvectorizer)? if so this ctor will show no default values
+template <size_t MaxNumTrainingItemsV>
+DocumentStatisticsEstimator<MaxNumTrainingItemsV>::DocumentStatisticsEstimator(
+    AnnotationMapsPtr pAllColumnAnnotations, 
+    size_t colIndex,
+    StringDecorator decorator, 
+    nonstd::optional<IndexMap> existingVocabulary, 
+    nonstd::optional<std::uint32_t> topKTerms, 
+    std::float_t minDf, 
+    std::float_t maxDf
+) :
+    BaseType(
+        std::move(pAllColumnAnnotations),
+        std::move(colIndex),
+        true,
+        std::move(decorator),
+        std::move(existingVocabulary),
+        std::move(topKTerms),
+        std::move(minDf),
+        std::move(maxDf)
+    ) {
 }
 
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator) :
+// ----------------------------------------------------------------------
+// |
+// |  Anonymous namespace
+// |
+// ----------------------------------------------------------------------
+namespace {
+
+inline Details::DocumentStatisticsTrainingOnlyPolicy::FrequencyMap PruneTermFreqMap(Details::DocumentStatisticsTrainingOnlyPolicy::FrequencyMap & termFrequency, 
+                                                                                    std::float_t minDf, 
+                                                                                    std::float_t maxDf, 
+                                                                                    std::float_t totalNumDocumentsFloat,
+                                                                                    nonstd::optional<std::uint32_t> topKTerms) {
+
+    using FrequencyMap = Details::DocumentStatisticsTrainingOnlyPolicy::FrequencyMap;
+    
+    //trim by minDf and maxDf
+    if (minDf > 0.0f || maxDf < 1.0f) {
+        for (FrequencyMap::iterator termFrequencyIter = termFrequency.begin(); termFrequencyIter != termFrequency.end();) {
```

Comment:
It is uncommon to have a for loop that doesn't increment the variable within the loop. I recommend that you convert this to a while loop.